### ISSUE TITLE
Ensure source table extraction can be run at IN2P3 CC

### ISSUE THREAD

**Issue by wmwv** [LSSTDESC/DC2-production#332]
The source table extraction code is written to run a NERSC using SLURM.

This issue is to track what would be necessary to make sure it can be run at IN2P3.
1. Are the README instructions clear enough to adapt?
2. Does the SLURM `taskfarmer` approach work at IN2P3 CC
   - My guess is that it doesn't and that `taskfarmer` is specific to NERSC.  But there is likely something similar for running a larger number of very similar jobs with no dependencies on each other.
3. Should we better organize things in `DC2-production/scripts`?

I'd like to start work on this issue once the version that runs at NERSC is merge #326 
so that this issue and branch can more cleanly focus on just the changes neeed to run at IN2P3 and resulting generalizations.

**Comment by wmwv** [comment]
@johannct Can you try and see if you can generate the Source Tables at IN2P3 CC using the instructions in the `DC2-production/scripts/README.md`?

There will be some obvious things to change, like the location of the data.  But I more want to know:
a) If something analogous to `taskfarmer` exists at IN2P3 CC and if it's easy to adapt?
b) Is there something less obvious that needs to be changed, or something that should be changed to make it easier to generalize to run at either NERSC or IN2P3 in a sustainable way.

You'll need the latest version of `gcr-catalogs`.  The source table catalog reader was just added last week.

**Comment by yymao** [comment]
And by the latest version Michael meant the master branch.

**Comment by johannct** [comment]
Why is `merge_source_cat.py` exposing GCR and a persistence to parquet file directly? `merge_tract_cat.py` only uses the butler and hdf5. Is it possible to keep these two stages consistent?

**Comment by wmwv** [comment]
Yes, the plan is to migrate `merge_tract_cat.py` to look much more like `merge_source_cat.py`, including writing to Parquet instead of HD5.

`merge_source_cat.py` uses GCR to get the list of required columns.

`merge_tract_cat.py` was getting _all_ of the columns.  In the future we will only get a subset of those columns, along the lines of the current `trim` files.

**Comment by johannct** [comment]
hmm I would rather not have to pull GCR and GCRCatalogs into the pipeline.

**Comment by yymao** [comment]
@johannct I am interested in knowing the reason so that we can come up with possible solutions.  I understand that catalog path issue that `GCRCatalogs` is facing on IN2P3, but `merge_source_cat.py` only uses the static method in the reader class, so the catalog data is not involved in the pipeline. Are there other issues that we should be aware of?

**Comment by johannct** [comment]
@yymao no strong reasons but a lot of new dependencies : GCR, GCRCatalogs, and then some parquet writer that is *not* in the DM ecosystem. When the butler is needed it makes sense to try to leverage the pipeline, but now the scripts start to mix two different software areas. Remember also that we went to some trouble to make sure that hdf5 was accessible through pandas with the DM conda system.

**Comment by johannct** [comment]
Isn't the requested columns a static piece of information? Is it really best to pull the whole GCR just for this information to be available?

**Comment by yymao** [comment]
Yes, and we use `GCRCatalogs` to maintain the set of the requested columns, and we think it's better to not duplicate the information so that we don't need to maintain two identical pieces. 

I do understand the concern of needing additional dependencies. This particular step of creating DPDD-like catalog is to bright the DM ecosystem with our own data access system, so I think that's the reason why it would involve both systems.

**Comment by johannct** [comment]
anyway, I can live with GCR and GCRCatalogs being local, like I do for metacal ngmix packages.
But the parquet reader needs to be shipped by DM. Else I am going to have to match a local package each time I switch to a new DM release, and for sure I will forget. So this is a blocker now.

**Comment by wmwv** [comment]
@johannct Yeah, I hear your concerns.  You're on (and have been on) the leading edge of what DM Science Pipelines hasn't quite done yet.

DM Science Pipelines creation of DPDD products will use Parquet files.  There is a signifiant bit of work that has been done but has not been merged into the main code.  We may choose, as DESC, to help this work to merge so that it does start to be shipped.

**Comment by johannct** [comment]
ok I guess for now I will disable the DPDD parts of the pipeline. I really do not want to mess with a local parquet writer. I leave it to you to move it up to DM management, as I do not know the concerned work. Should I go on with the object summary HDF5 files, or does that complicate your life rather than simplify it?

**Comment by yymao** [comment]
We are hoping to provide the analysis team the object summary catalogs in a shorter time scale so I think we should go ahead with the current `merge_tract_cat.py` with HDF5. We do need to think about the path forward if we can't have a script that uses `butler` and `parquet` at the same time.

**Comment by johannct** [comment]
I proposed a pull request for the other script, if we can live for now with something like the merge_tract_cat. Up to you (it does not invalidate a GCR/parquet use if available)

**Comment by johannct** [comment]
the object summary files have been shipped for the recent 6mths transfer

**Comment by yymao** [comment]
Ah so you were asking if we should make the source summary catalogs with HDF5. That I am not sure because I don't know how many analysis teams use source summary catalogs and whether it's worth the time to modify the reader for it.

**Comment by johannct** [comment]
yes, I presume that you will modify merge_tract_cat.py to use GCR/parquet rather than butler/hdf5 *when*parquet will be available in DM.  But merge_source_cat.py already uses GCR/parquet and I was under the impression that it is worth running, at least for the 6 months processing. Probably a question/confirmation for a larger audience.

**Comment by wmwv** [comment]
Agreed.  Object Table for now is fine.  Thanks very much for doing this as part of the 6-month processing.
